### PR TITLE
Uses the 'new' search suggestion API for the main app search

### PR DESCRIPTION
**Pull request by orta**
Fixes DISCO-550

Moves us to the new suggest API, and conservatively adds support for all the new suggestion types. Note that this code is not aiming for pure elegance, it was built to make it easy to remove the `SearchResult` class etc when Local Discovery is shipped and the SearchVC only has to handle one type of data (suggestions)

☎️

### CODE REVIEW

**Comment by orta** [comment]
Articles can be found in the app again

screen shot 2018-12-07 at 5 58 00 pm

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Artsy/Networking/API_Modules/ArtsyAPI+Search.m [hunk 1]
```diff
@@ -42,14 +43,32 @@ + (AFHTTPRequestOperation *)searchWithFairID:(NSString *)fairID andQuery:(NSStri
         NSArray *jsonDictionaries = JSON;
         NSMutableArray *returnArray = [NSMutableArray array];
 
-        for (NSDictionary *dictionary in jsonDictionaries) {
-            if ([SearchResult searchResultIsSupported:dictionary]) {
+        if (fairID) {
+            // Old style obj -> objc class mapping for fairs
```
Comment (alloy) [inline Artsy/Networking/API_Modules/ArtsyAPI+Search.m]: Can you extend this comment to explain why the old style is needed for fairs?

File: Artsy/Networking/ARRouter.m [hunk 2]
```diff
@@ -855,7 +855,8 @@ + (NSURLRequest *)newArtworkInfoRequestWithID:(NSString *)artworkID
 
 + (NSURLRequest *)newSearchRequestWithQuery:(NSString *)query
 {
-    return [self requestWithMethod:@"GET" path:ARNewSearchURL parameters:@{ @"term" : query }];
+    return [self requestWithMethod:@"GET" path:ARNewSearchURL parameters:@{ @"term" : query, @"visible_to_public": @(YES), @"agg": @(NO) }];
+//    visible_to_public=true&fair_id=&size=7&agg=false&term=ssd
```
Comment (alloy) [inline Artsy/Networking/ARRouter.m]: Meant to leave in?

File: Artsy/View_Controllers/Search/ARSearchViewController.m [hunk 3]
```diff
@@ -413,9 +414,17 @@ - (void)selectedResult:(SearchResult *)result ofType:(NSString *)type fromQuery:
 
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    SearchResult *result = [self.searchDataSource objectAtIndex:indexPath.row];
-    NSString *type = [NSStringFromClass([result.model class]) lowercaseString];
-    [self selectedResult:result ofType:type fromQuery:self.textField.text];
+    NSObject <SearchResultable >*result = [self.searchDataSource objectAtIndex:indexPath.row];
```
Comment (alloy) [inline Artsy/View_Controllers/Search/ARSearchViewController.m]: Small whitespace issue.

**Pull request merged**